working/templates/userdefinedAPIsolution/$SCRIPTNAME$_1/$SCRIPTNAME$_1.cs

```diff
@@ -9,6 +9,35 @@ namespace $NAMESPACE$_1
 	/// </summary>
 	public class Script
 	{
+		private static IDictionary<string, string> GetParametersFromInput(ApiTriggerInput requestData)
+		{
+			var parameters = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+		
+			if (requestData != null)
+			{
+				var queryList = requestData.QueryParameters?.GetAllKeys();
+				if (requestData.Parameters?.Count > 0 && (queryList == null || queryList.Count == 0))
+				{
+					foreach (var parameter in requestData.Parameters)
+					{
+						parameters[parameter.Key] = parameter.Value;
+					}
+				}
+				else if (queryList?.Count > 0)
+				{
+					foreach (var query in queryList)
+					{
+						if (requestData.QueryParameters.TryGetValue(query, out var value))
+						{
+							parameters[query] = value;
+						}
+					}
+				}
+			}
+		
+			return parameters;
+		}
+		
```

Comment:
As the QueryParameters are only available from DM 10.4.1 onwards, this would restrict a lot of people. I would suggest to remove this part and maybe even the previous change and instead add a link to DataMiner docs in a comment. On DataMiner docs is a page with what is available and what it does: [Defining a new API](https://docs.dataminer.services/user-guide/Advanced_Modules/User-Defined_APIs/Defining_an_API/UD_APIs_Defining_a_new_API.html)

Reply:
Best to use an aka link as well so that if the docs page moves, we don't need to update the template.